[docs] Update notifications sample code (#28095)
expo/expo: docs/pages/push-notifications/push-notifications-setup.mdx

The code below shows a working example of how to register for, send, and receive push notifications in a React Native app. Copy and paste it into your project:

> ```jsx App.tsx

Author: Good catch

});
/* @end */

> /* @info Sends a notification to Expo. Expo's push notification tool at https://expo.dev/notifications may also be used. */

Reviewer: "Sends a notification to Expo" feels a bit unclear, but maybe it's just me. I'd prefer to clarify that we're sending something to Expo, and then Expo is sending that along to a recipient device. 

Maybe I'm overthinking this, but when I first onboarded to Push Notifications I did the demo and I didn't fully grasp that clicking the big blue "Send Notification" button actually sent anything anywhere. It seemed all local, like I was triggering a cute little demo notification to my app, from my app, all within my device. (And thus it felt a bit gimmicky, tbh). 

Anyway, I just feel like we could clarify that we're not sending a notification to Expo, but that Expo is sending a notification to _us_ (or whatever device we specify). 

Feel free to disregard this, sorry I don't have a specific suggestion on the copy.

> /* @info Sends a notification to Expo. Expo's push notification tool at https://expo.dev/notifications may also be used. */

Reviewer 2: This is good context @christopherwalter!

Maybe we can rephrase this and just say "Sends a notification to the device. Can also use Expo's push notification tool to test sending a notification: https://expo.dev/notifications." 

I don't have much context on how this works here but we can also rephrase it to "Sends a notification to the device from Expo." as per your explanation. This way, the comment here will be clear about what action the function in subject performs.

      lightColor: '#FF231F7C',
    });
  }

  if (Device.isDevice) {
    const { status: existingStatus } =
      await Notifications.getPermissionsAsync();
    let finalStatus = existingStatus;
    if (existingStatus !== 'granted') {
      const { status } = await Notifications.requestPermissionsAsync();
      finalStatus = status;
    }
    if (finalStatus !== 'granted') {
      handleRegistrationError('Permission not granted to get push token for push notification!');
      return;
    }
    const projectId =
      Constants?.expoConfig?.extra?.eas?.projectId ??
      Constants?.easConfig?.projectId;
    if (!projectId) {
      handleRegistrationError('Project ID not found');
    }
    try {

>     /* @info This provides the ExpoPushToken using the unique project ID. */

Reviewer: IMO this could be clearer by saying something like `This fetches the ExpoPushToken (if not previously fetched), which is unique to this device and projectID`.

Author: ```suggestion
    /* @info This fetches the ExpoPushToken (if not previously fetched), which is unique to this device and projectID. */
```

      const pushTokenString = (
        await Notifications.getExpoPushTokenAsync({
          projectId,
        })
      ).data;
    /* @end */
      console.log(pushTokenString);
      return pushTokenString;

>     } catch (e: any) {

Reviewer: nit: can we avoid `any`?

Author: @christopherwalter I'll have a look to see if that is possible. The error isn't always a string as I recall.

>       handleRegistrationError(e);

Author: ```suggestion
      handleRegistrationError(`${e}`);
```

    }
  } else {
    handleRegistrationError('Must use physical device for push notifications');
  }
}

export default function App() {
  const [expoPushToken, setExpoPushToken] = useState('');
  const [notification, setNotification] = useState<
    Notifications.Notification | undefined
  >(undefined);
  const notificationListener = useRef<Notifications.Subscription>();
  const responseListener = useRef<Notifications.Subscription>();

  useEffect(() => {

>     /* @info Gets the push token and shows it in the UI, or shows the error if one occurs */

Reviewer 2: ```suggestion
    /* @info Gets the push token and displays it in the UI, or in case of an error, displays the error message. */
```

    registerForPushNotificationsAsync()
      .then((token) => setExpoPushToken(token ?? ''))
      .catch((error: any) => setExpoPushToken(`${error}`));
    /* @end */

    notificationListener.current =
      Notifications.addNotificationReceivedListener((notification) => {
        setNotification(notification);
      });

    responseListener.current =
      Notifications.addNotificationResponseReceivedListener((response) => {
        console.log(response);
      });

    return () => {
      notificationListener.current &&
        Notifications.removeNotificationSubscription(
          notificationListener.current,
        );
      responseListener.current &&
        Notifications.removeNotificationSubscription(responseListener.current);
    };
  }, []);

  return (
    <View
      style={{ flex: 1, alignItems: 'center', justifyContent: 'space-around' }}
    >

>       <Text>Your expo push token: {expoPushToken}</Text>

Author: ```suggestion
      <Text>Your Expo push token: {expoPushToken}</Text>
```

>       <Text>Your expo push token: {expoPushToken}</Text>

Author: I've attempted to make these more consistent.
